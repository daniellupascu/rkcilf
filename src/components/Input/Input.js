import React from "react";
import PropTypes from "prop-types";

import {
  InputWrapper,
  Label,
  InputContainer,
  InputStyled,
} from "./InputStyled";

const Input = ({ value, label, placeholder, onChange }) => {
  return (
    <InputWrapper>
      <Label>{label}</Label>
      <InputContainer>
        <InputStyled
          value={value}
          onChange={onChange}
          placeholder={placeholder}
        />
      </InputContainer>
    </InputWrapper>
  );
};

export default Input;

Input.protoTypes = {
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
