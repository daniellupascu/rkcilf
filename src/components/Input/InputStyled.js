import styled from "styled-components";

export const InputWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

export const Label = styled.div`
  margin: 5px;
  font-size: 12px;
`;

export const InputContainer = styled.div`
  display: flex;
  align-items: center;
  height: 40px;
  width: 350px;
  margin: 5px;
  position: relative;
`;

export const InputStyled = styled.input`
  padding: 0 20px 0 5px;
  align-self: stretch;
  width: 100%;

  border: 1px solid #cecece;
  border-radius: 5px;

  &:focus {
    outline-color: #c179b9;
  }
`;

export const ExpandIconContainer = styled.div`
  right: 8px;
  position: absolute;
  cursor: pointer;
`;
