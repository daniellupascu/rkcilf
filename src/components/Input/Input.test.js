import React from "react";
import renderer from "react-test-renderer";
import Input from "./Input";

describe("Test Input Component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Input label="Name" placeolder="Name" onChange={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
