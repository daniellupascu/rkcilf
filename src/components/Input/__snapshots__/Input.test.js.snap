// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component renders correctly 1`] = `
<div
  className="sc-AxjAm jsGdgg"
>
  <div
    className="sc-AxirZ hEggWj"
  >
    Name
  </div>
  <div
    className="sc-AxiKw cMDocJ"
  >
    <input
      className="sc-AxhCb jVuUNY"
      onChange={[Function]}
    />
  </div>
</div>
`;
